.drone.yml

```diff
@@ -80,372 +77,340 @@ steps:
-      aws_secret_access_key:
-        from_secret: aws_secret_access_key
-      terraform_tf_states_bucket_name:
-        from_secret: terraform_tf_states_bucket_name
-      vsphere_server:
-        from_secret: vsphere_server
-      vsphere_password:
-        from_secret: vsphere_password
-      vsphere_user:
-        from_secret: vsphere_user
-      dockerhub_username:
-        from_secret: dockerhub_username
-      dockerhub_password:
-        from_secret: dockerhub_password
-
-  - name: e2e
-    # KUBECTL 1.25.3 - KUSTOMIZE 3.5.3 - HELM 3.1.1 - YQ 4.21.1 - ISTIOCTL 1.9.4 - FURYCTL 0.9.0 - BATS 1.1.0
-    image: quay.io/sighup/e2e-testing:1.1.0_0.9.0_3.1.1_1.9.4_1.25.3_3.5.3_4.21.1
+    commands:
+      - sh ./katalog/tests/kind/generate-template.sh 1.26.6
+    environment:
+      KUBERNETES_VERSION: 1.26.6
+  - name: Create Kind Cluster
+    image: quay.io/sighup/dind-kind-kubectl-kustomize:0.20.0_1.29.1_3.10.0
     pull: always
-    volumes:
-      - name: shared
-        path: /shared
-    depends_on: [init]
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-1.26.6
+      KIND_CONFIG: ./config-$${CLUSTER_NAME}.yaml
+      CLUSTER_VERSION: 1.26.6
+      KUBECONFIG: ./kubeconfig-$${CLUSTER_NAME}
     commands:
-      - export CLUSTER_NAME=124
-      - export INSTANCE_IP=$(cat /shared/machine/ip)
-      - export KUBECONFIG=/shared/kube/kubeconfig-125
-      - bats -t katalog/tests/tests.bats
-
-  - name: destroy
-    image: quay.io/sighup/e2e-testing-drone-plugin:v1.25.3
+      - kind create cluster --config $${KIND_CONFIG}
+      - kind get kubeconfig --name $${CLUSTER_NAME} > $${KUBECONFIG}
+    depends_on:
+      - Generate Kind Config file
+  - name: End-to-End Tests
+    image: quay.io/sighup/e2e-testing:1.1.0_0.11.0_3.1.1_1.9.4_1.26.3_3.5.3_4.33.3
     pull: always
-    depends_on: [e2e]
-    settings:
-      action: destroy
-      pipeline_id: cluster-125
-      aws_default_region:
-        from_secret: aws_region
-      aws_access_key_id:
-        from_secret: aws_access_key_id
-      aws_secret_access_key:
-        from_secret: aws_secret_access_key
-      terraform_tf_states_bucket_name:
-        from_secret: terraform_tf_states_bucket_name
-      vsphere_server:
-        from_secret: vsphere_server
-      vsphere_password:
-        from_secret: vsphere_password
-      vsphere_user:
-        from_secret: vsphere_user
-      dockerhub_username:
-        from_secret: dockerhub_username
-      dockerhub_password:
-        from_secret: dockerhub_password
+    network_mode: host
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-1.26.6
+      KUBECONFIG: ./kubeconfig-$${CLUSTER_NAME}
+    depends_on:
+      - Create Kind Cluster
+    commands:
+      - export INTERNAL_PORT=$(grep 'internalPort' ./env-$${CLUSTER_NAME}.yaml | awk '{ print $2 }')
+      - export EXTERNAL_PORT=$(grep 'externalPort' ./env-$${CLUSTER_NAME}.yaml | awk '{ print $2 }')
```

Comment:
Instead of `grep`ing and `awk`ing you can create a `.env` file instead of a YAML one with the script and replace these 2 lines with `source ./env-$${CLUSTER_NAME}}.env`.

This simplifies the CI and also reduces the chance of an error.